docs/us.md

```diff
@@ -0,0 +1,225 @@
+An update proposal continues being _active_ until it becomes _decided_, it may be _decided_ to approve
+or to reject.
+If a proposal is rejected then it doesn't affect any consensus rules and we don't consider it anymore.
+Note: rejected proposal may become active again if rollback occurs.
+
+* **Confirmed**  
+If a proposal has been approved in some block and there are at least `k` blocks after this one, then
+update proposal is called _confirmed_.
+_Confirmed_ state reflects the fact that a proposal's state cannot be changed anymore 
+because we have guarantee that at most `k` blocks may be rolled back.
+
+### Software and block versions. Data associated with block version
+
+When a developer wants to fix some bug or add some features to application,
+which don't affect protocol rules, he should propose an update with new `SoftwareVersion`.
+```
+data SoftwareVersion = SoftwareVersion
+    { svAppName :: !ApplicationName
+    , svNumber  :: !NumSoftwareVersion
+    }
+```
+
+`SoftwareVersion` contains application name and numeric representation of application version.
+
+If a developer wants to change some protocol rules, for example, `Address` format, 
+he must propose update with new `BlockVersion`
+
+```
+data BlockVersion = BlockVersion
+    { bvMajor :: !Word16
+    , bvMinor :: !Word16
+    , bvAlt   :: !Word8
+    }
+```
+
+So block version essentially is a tuple `(Maj, Min, Alt)`.
+Block version is also called **protocol version**.
+
+There are some protocol constants which depend on `BlockVersion`, for example max block size, slot duration, etc.
+Update system was designed in such way that you must attach new values of this constants if they are updated.
+`BlockVersionModifier` reflects these changes:
+```
+data BlockVersionModifier = BlockVersionModifier
+    { bvmScriptVersion     :: !(Maybe ScriptVersion)
+    , bvmSlotDuration      :: !(Maybe Millisecond)
+    , bvmMaxBlockSize      :: !(Maybe Byte)
+    , bvmMaxHeaderSize     :: !(Maybe Byte)
+    , bvmMaxTxSize         :: !(Maybe Byte)
+    , bvmMaxProposalSize   :: !(Maybe Byte)
+    , bvmMpcThd            :: !(Maybe CoinPortion)
+    , bvmHeavyDelThd       :: !(Maybe CoinPortion)
+    , bvmUpdateVoteThd     :: !(Maybe CoinPortion)
+    , bvmUpdateProposalThd :: !(Maybe CoinPortion)
+    , bvmUpdateImplicit    :: !(Maybe FlatSlotId)
+    , bvmSoftforkRule      :: !(Maybe SoftforkRule)
+    , bvmTxFeePolicy       :: !(Maybe TxFeePolicy)
+    , bvmUnlockStakeEpoch  :: !(Maybe EpochIndex)
+    }
+```
+So if a value is going to be updated then a field is `Just` and `Nothing` otherwise.
+
+### Adoption of block version
+
+Assume a proposal which is bumping `BlockVersion` became _confirmed_.
+Though the proposal is already _confirmed_ along with block version but stakeholders
+haven't updated its software yet, hence, they can't validate block of _confirmed_ block version.
+
+To avoid a situation when adversary stakeholder issues invalid block of just _confirmed_ block version,
```

Comment:
For block versions we use term **competing** rather than **confirmed** to avoid ambiguity of term **confirmed**.